### PR TITLE
sigstore: v3 bundles

### PR DESCRIPTION
**Pull request by woodruffw**
Bumps `sigstore-protobuf-specs`, giving us access to the latest Bundle version.

### CODE REVIEW

**Comment by woodruffw** [comment]
I think this is good to go. It adds support for verifying v3 bundles, but still emits v2 ones. 

In terms of switching signing to v3: I think the 3.x release series would be reasonable.

File: sigstore/verify/models.py [hunk 1]
```diff
@@ -317,22 +318,22 @@ def from_bundle(
...
-        if bundle.media_type == _BUNDLE_0_1:
+        if media_type == KnownBundleType.BUNDLE_0_1:
             if not inclusion_promise:
                 raise InvalidMaterials("bundle must contain an inclusion promise")
             if inclusion_proof and not inclusion_proof.checkpoint.envelope:
                 logger.debug(
                     "0.1 bundle contains inclusion proof without checkpoint; ignoring"
                 )
-        elif bundle.media_type == _BUNDLE_0_2:
+        elif media_type == KnownBundleType.BUNDLE_0_2:
```
Comment (jku) [inline sigstore/verify/models.py:336]: what about `BUNDLE_0_3` here?

Comment (woodruffw) [reply sigstore/verify/models.py:336]: Whoops, thanks.

**Review by jku — APPROVED** [review]
Seems quite reasonable as is. Some comments : 
* tests would be nice but I can see how they become more viable once the signing works too
* `KnownBundleType` is a bit of a mouthful: it could probably be just `BundleType`

**Comment by woodruffw** [comment]
> * `KnownBundleType` is a bit of a mouthful: it could probably be just `BundleType`

Yeah, I like that 🙂 -- I'll make a quick change (and also update the CHANGELOG).

**Review by jku — APPROVED** [review]

**Pull request merged**